templates/deployments/appname/app.jsonnet.tpl

```diff
@@ -62,19 +62,31 @@ local all = {
 	namespace: ok.Namespace(app.namespace) {
 		metadata+: {
 			annotations+: {
-				'iam.amazonaws.com/permitted': '%s_service_role' % app.name,
+        {{- if stencil.Arg "aws.useKIAM" }}
+        'iam.amazonaws.com/permitted': '%s_service_role' % app.name,
+        {{- end }}
 			},
 			labels+: sharedLabels,
 		},
 	},
+  svc_acct: ok.ServiceAccount('%s-svc' % app.name, app.namespace) {
+    metadata+: {
+      labels+: sharedLabels,
+      annotations+: {
+        {{- if stencil.Arg "aws.useKIAM" }}
+        'eks.amazonaws.com/role-arn': 'arn:aws:iam::{{ .Runtime.Box.AWS.DefaultAccountID }}:role/%s-%s' % [app.bento, app.name]
```

Comment:
Pretty sure this `eks.amazonaws.com/role-arn` is for IRSA. 
KIAM would look more like `iam.amazonaws.com/role` right?